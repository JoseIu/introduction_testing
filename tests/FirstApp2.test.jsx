import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp/>", () => {
	const title = "Hola, Soy Goku";
	const subTitle = "Soy un subtitulo";
	test("Should to do match with snapshot ", () => {
		const { container } = render(<FirstApp title={title} />);

		expect(container).toMatchSnapshot();
	});

	test('Should show a message "Hola, Soy Goku"', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
	test("Title should show in a H1 ", () => {
		render(<FirstApp title={title} />);

		expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
			title
		);
	});
	test("Should show subtitle sended by PROPS ", () => {
		render(<FirstApp title={title} subTitle={subTitle} />);

		expect(screen.getAllByText(subTitle).length).toBe(2);
	});
	test("The subTitle should be in a H2 ", () => {
		render(<FirstApp title={title} subTitle={subTitle} />);

		const headings = screen.getAllByRole("heading", { level: 2 });

		headings.forEach((heading) =>
			expect(heading.innerHTML).toContain(subTitle)
		);
	});
});
