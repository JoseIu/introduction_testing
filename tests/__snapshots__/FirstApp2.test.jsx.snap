// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Should to do match with snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <h2>
    No hay subtítulo
  </h2>
  <h2>
    No hay subtítulo
  </h2>
  <p>
    Fernando Herrera
  </p>
</div>
`;
